### PR TITLE
fix: multiselect in position work now

### PR DESCRIPTION
**Pull request by mr-fox93**
- Added `handleButtonClick` function in Filters.tsx to enable multi-select functionality for position selection.
- Updated `jobSpecializationFilter` from `string` to `string[]` in FilterContext to support multiple selections.
- Modified `filterByPosition` in filters.ts to accommodate the new multi-select feature.

### CODE REVIEW

**Review by szymonkadas — APPROVED** [review]
Seems fine, dunno about profile-data.ts, ig it should be deleted entirely?

**Comment by mr-fox93** [comment]
> Seems fine, dunno about profile-data.ts, ig it should be deleted entirely?

line 1-28 is important, I commented the code, which is probably no longer used, because we now show profiles from the database everywhere, not stock ones. @Sumick  take a look please ?

**Review by wawrzonnn — APPROVED** [review]
Could you please remove the profile-data file as it seems unnecessary? Just ensure to move the 'enum JobSpecialization' we're using to our frontend types.

**Review by mmcode9407 — APPROVED** [review]

File: src/components/ProfileList/profile-data.ts [hunk 1]
```diff
@@ -28,38 +28,38 @@ export enum JobSpecialization {
...
+//     remote: 'Remote',
+//     technology: ['Javascript', 'React', 'Vue.js'],
+//   },
+//   {
+//     id: 2,
+//     name: 'Kristin Watson',
+//     seniority: 'Junior',
+//     jobSpecialization: JobSpecialization.Frontend,
+//     others: '+5 more',
+//     country: 'Poland',
```
Comment (Sumick) [inline src/components/ProfileList/profile-data.ts:49]: **nit:** Why comment? If commenting not removing, let's add explanation in code why it's not removed

**Review by Sumick — CHANGES_REQUESTED** [review]
Let's add this landing fix for negative values when selecting from DB to this PR before merging, because app cannot be deployed now

**Comment by mr-fox93** [comment]
> Let's add this landing fix for negative values when selecting from DB to this PR before merging, because app cannot be deployed now

i think now should work :

export async function getRandonProfiles(profilesCount: number) {
  const totalProfiles = await prisma.profile.count()

  if (profilesCount >= totalProfiles) {
    return getPublishedProfilesPayload()
  }

  const maxSkipValue = totalProfiles - profilesCount
  const skipValue =
    maxSkipValue > 0 ? Math.floor(Math.random() * maxSkipValue) : 0

  const randomRecords = await prisma.profile.findMany({
    take: profilesCount,
    where: {
      state: PublishingState.APPROVED,
    },
    orderBy: {
      id: 'asc',
    },
    skip: skipValue,
    include: {
      user: {
        include: {
          githubDetails: true,
        },
      },
      country: true,
      city: true,
    },
  })

  return randomRecords.map(serializeProfileToProfileModel)
}

File: src/components/ProfileList/profile-data.ts (hunk 1 above)
Comment (mr-fox93) [reply src/components/ProfileList/profile-data.ts]: deleted, it was a fragment used when we did not download profiles from the database, I guess.

File: src/backend/profile/profile.service.ts [hunk 2]
```diff
@@ -206,14 +206,25 @@ export async function getProfileByUserEmail(email: string) {
 }
 
 export async function getRandonProfiles(profilesCount: number) {
```
Comment (Sumick) [inline src/backend/profile/profile.service.ts:208]: **nit:** `getRandomProfiles*` you have typo `n`

**Review by Sumick — APPROVED** [review]
Looks good!

**Pull request merged**